default/python/AI/freeorion_tools/_freeorion_tools.py

```diff
@@ -428,25 +428,26 @@ def get_named_real(name: str) -> float:
     return 1.0
 
 
-def get_game_rule_int(name: str) -> int:
+def get_game_rule_int(name: str, default: int = 0) -> int:
     """
-    Returns a integer value for a game rule.
-    Exceptions are handled like in get_named_int.
+    Returns an integer value for a game rule.
+    If the current game does not include the rule, the default value is returned.
+    Note that unlike named values, which should always exists unless someone changes the scripting without adapting
+    the AI, game rules are set when a game is started. When loading old game with a newer version, the scripting
+    will use the default values of game rules not found in the save file.
     """
     rules = fo.getGameRules()
     if rules.ruleExistsWithType(name, fo.ruleType.int):
         return rules.getInt(name)
-    error(f"Requested integer rule {name} does not exist!")
-    return 1
+    return default
 
 
-def get_game_rule_real(name: str) -> int:
+def get_game_rule_real(name: str, default: int = 0.0) -> float:
```

Comment:
Should not we preserve the old default value?
```suggestion
def get_game_rule_real(name: str, default: int = 1.0) -> float:
 ```